eos_data_distribution/ndn/http.py

```diff
@@ -99,25 +111,31 @@ def _got_stream(self, msg, task, data):
 
     def _got_buf(self, data, buf):
         data.setContent(buf)
-        self.sendFinish(data)
+        self.emit('data', data)
```

Comment:
(I haven’t profiled but…) I suspect emitting a signal for every chunk of data received is going to add considerable CPU overhead. GObject signal emissions are not cheap.